packages/lmi/tests/test_rpm.py

```diff
@@ -0,0 +1,148 @@
+"""Tests for request rate limit (RPM) functionality in LiteLLMModel.
+
+This module contains tests to verify the rate limiting functionality,
+including both sequential and concurrent request scenarios.
+"""
+
+import asyncio
+import time
+
+import pytest
+
+from lmi.llms import CommonLLMNames, LiteLLMModel
+from lmi.types import Message
+
+
+def _setup_model_config(rpm_limit: int | None) -> tuple[dict, str]:
+    """Set up the model configuration based on RPM limit.
+
+    Args:
+        rpm_limit: The RPM limit to test with. None means no limit.
+
+    Returns:
+        Tuple containing the model configuration and test name.
+    """
+    model_config = {
+        "model_list": [
+            {
+                "model_name": CommonLLMNames.OPENAI_TEST.value,
+                "litellm_params": {
+                    "model": CommonLLMNames.OPENAI_TEST.value,
+                    "temperature": 0.1,
+                    "max_tokens": 4096,
+                },
+            }
+        ]
+    }
+
+    if rpm_limit is not None:
+        model_config["request_limit"] = {
+            "deepseek/deepseek-chat": f"{rpm_limit}/minute"
+        }  # type: ignore[assignment]
+
+    return model_config, f"RPM={rpm_limit}" if rpm_limit else "No Limit"
+
+
+def _validate_rpm_results(results: dict[str, float]) -> None:
```

Comment:
So in unit testing, it's best to avoid helper functions, if they're only used once.

The idea is unit tests should have "locality of behavior" which means they can be read in one place, without reading helper functions.

It's okay if a unit test is a bit longer, if you're hitting line length issues from linters just disable them